array_methods_assignment.js

```diff
@@ -0,0 +1,519 @@
+  return arr.every((num) => {
+  if (num > 0 && (num % 2) === 0 && (num % 10) !== 0) {
+    return true;
+    }
+  })
+}
+
+let strictNum = posi(arr11)
+let strictNum1 = posi(arr11b)
+
+console.log(strictNum)
+console.log(strictNum1)
+
+// 13. Write a function that returns the sum of all odd numbers in an Array
+//
+// Hint: What is the second argument in `reduce`?
+//
+// ```
+// Input: [2,3,4,5]
+// Output: 8
+//
+// Input: [1,2,3,4,5]
+// Output: 9
+// ```
+console.log("13._______________________________________");
+
+let arr12 = [33, 34, 35, 36]
+let arr12b = [2, 3, 4, 5, 6, 7]
+
+function oddSums(arr){
+  return arr.reduce((total, currentElem) => {
+    if(currentElem % 2 !== 0) {
+    total += currentElem
+  }
+  return total
+}, 0)
+}
+
+
+let addOdd = oddSums(arr12)
+let addOddb = oddSums(arr12b)
+
+console.log(addOdd)
+console.log(addOddb)
+
+// 14. Write a function that returns a new array containing all of the strings in the original array uppercased.
+//
+// ```
+// Input: [{}, 1, "cat", 3, ["hi"], {name: "dog"}, "dog", "bear"]
+// Output: ["CAT", "DOG", "BEAR"]
+// ```
+
+console.log("14._______________________________________");
+
+let mixedArr = [{}, 1, "cat", 3, ["hi"], {name: "dog"}, "dog", "bear"];
+
+function takeString(arr) {
+  function makeItBig(str) {
+  let big = str.toUpperCase();
+  return big;
+}
+  return arr.filter((string) => {
+    if( typeof string === "string") {
+    return string;
+    }
+  }).map(makeItBig)
+}
+
+
+let newArr = takeString(mixedArr);
+
+console.log(newArr)
+
+// 1. Using Array.prototype add a new method `myForEach` that mimics the behavior of the built in `forEach`.
+
+console.log("Bonus 1. _______________________________________");
+
+let myArr = [1,2,3,4]
+
+Array.prototype.myForEach = function(x) {
+  for (let i of this){
+    x(i)
+  }
+};
+
+myArr.myForEach(console.log);
+
+// 2. Use your `myForEach` to add `myMap` to the Array prototype. `myMap` should behave the same as regular `map`.
+
+console.log("Bonus 2._______________________________________");
+
+Array.prototype.myMap = function (x) {
+  let newArr = [];
+  for (let i of this) {
+    newArr.push(x(i))
+  }
+  return newArr;
+};
+
+console.log(myArr.myMap((num) => num * 2));
+
+// 3. Write a `myFilter` that behaves the same as `filter`.
+
+console.log("Bonus 3._______________________________________");
+
+Array.prototype.myFilter = function (x) {
+  let newArr = [];
+  for (let i of this) {
+    if (x(i)){
+      newArr.push(i)
+    }
+  }
+  return newArr;
+}
+
+console.log(myArr.myFilter((num) => num % 2 === 0))
+
+// 4. Write a `myEvery` that behaves the same as `every`.
+
+console.log("4._______________________________________");
+
+let trueArr = [6, 7, 8, 9];
+let falseArr = [6, 7, 8, 9, 4];
+
+Array.prototype.myEvery = function (x) {
+  let falseNum = 0;
+  for (let i of this) {
+    if (x(i)) {
+    } else {
+      falseNum += 1;
+    }
+  } if (falseNum > 0) {
+    return false;
+  } else {
+    return true;
+  }
+}
```

Comment:
A little confused about this answer.. The every method checks to make sure that every element meets the condition. If even one doesn't it returns the boolean false.